File: frame/src/main/java/com/yumore/frame/basic/BaseContract.java
package com.yumore.frame.basic;

import io.reactivex.disposables.Disposable;

/**
 * BasePresenter
 *
 * @author Nathaniel
 * nathanwriting@126.com
 * @version v1.0.0
 * @date 2018/3/8 - 15:02
 */
public interface BaseContract<V extends BaseView> {
    @SuppressWarnings("unchecked")
    void attachView(V baseView);

    void detachView();

    void addDisposable(Disposable disposable);

    void unDisposable();
}

File: basicui/src/main/java/com/nathaniel/baseui/binding/BaseContract.kt
package com.nathaniel.baseui.binding

import com.valuelink.common.basic.BaseViewer
import io.reactivex.disposables.Disposable

/**
 * BasePresenter
 *
 * @author Nathaniel
 * @contact <a href="mailto:nathanwriting@126.com">contact me</a>
 * @version v1.0.0
 * @date 2018/3/8 - 15:02
 */
interface BaseContract<V : BaseViewer> {
    fun attachView(baseViewer: V)
    fun detachView()
    fun addDisposable(disposable: Disposable?)
    fun unDisposable()
    fun setLoadable(loadable: Boolean)
}